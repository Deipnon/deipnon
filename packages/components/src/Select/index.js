// @flow

import * as React from 'react'
import { css } from '@emotion/core'
import styled from '@emotion/styled'
import theme from '@deipnon/themes/lib'

const inputBaseStyle = css`
	display: block;
	padding: ${theme.styles.formPaddings};
	font-size: ${theme.styles.formFontSize};
	font-weight: ${theme.styles.formFontWeight};
	line-height: ${theme.styles.formLineHeight};
	color: ${theme.styles.formColor};
	background-color: ${theme.styles.formBgColor};
	background-clip: padding-box;
	border: ${theme.styles.formBorderWidth} solid ${theme.styles.formBorderColor};
	height: ${theme.styles.formHeight};
`
const withBlock = ({ block }) => {
	if (block) {
		return css`
			width: 100%;
		`
	}
}

const SelectElement = styled('select')(inputBaseStyle, withBlock)

type PropsType = {|
	block: ?boolean,
	children: React$Node,
	onChange: (e: SyntheticEvent<HTMLSelectElement>) => void,
	value: ?string
|}

const Select = (props: PropsType) => (<SelectElement {...props} >
	{props.children}
</SelectElement>)

Select.defaultProps = {
	block: false
}

export default Select
