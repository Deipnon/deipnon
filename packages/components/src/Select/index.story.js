import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, boolean } from '@storybook/addon-knobs/react'

import Select from '.'

const stories = storiesOf('Select', module)
stories.addDecorator(withKnobs)

class SelectListSample extends React.Component {
	constructor (props) {
		super(props)

		this.state = {
			selectedValue: 2,
			selectListItems: [{
				text: 'Item 1',
				value: 1
			}, {
				text: 'Item 2',
				value: 2
			}]
		}
	}

	onSelectChange = (e) => {
		const newValue = e.currentTarget.value
		this.setState({
			selectedValue: newValue
		})
	}

	render () {
		const { selectListItems, selectedValue } = this.state
		return (
			<Select value={selectedValue} onChange={this.onSelectChange} block={boolean('Block', true)}>
				{selectListItems.map((item, index) => (
					<option key={index} value={item.value}>{item.text}</option>
				))}
			</Select>
		)
	}
}

stories.add('default', () => <SelectListSample />)
