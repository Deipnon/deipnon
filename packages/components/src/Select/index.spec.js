import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-emotion'

import Select from '.'

describe('Select', () => {
	test('renders correctly', () => {
		const tree = renderer.create(<Select onChange={() => {}}>
			<option value="1">Test</option>
		</Select>).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
