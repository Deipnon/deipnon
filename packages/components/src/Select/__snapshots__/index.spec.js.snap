// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<select
  className="css-b1s8qf"
  onChange={[Function]}
>
  <option
    value="1"
  >
    Test
  </option>
</select>
`;
